custom_components/unifiprotect/camera.py

```diff
@@ -139,26 +194,59 @@ class UnifiProtectCamera(UnifiProtectEntity, Camera):
 
     def __init__(
         self,
-        upv_object,
-        protect_data,
-        server_info,
-        camera_id,
-        disable_stream,
+        protect: ProtectApiClient,
+        protect_data: UnifiProtectData,
+        camera: UnifiCamera,
+        channel: CameraChannel,
+        is_default: bool,
+        secure: bool,
+        disable_stream: bool,
     ):
         """Initialize an Unifi camera."""
-        super().__init__(upv_object, protect_data, server_info, camera_id, None)
-        self._name = self._device_data["name"]
+        super().__init__(protect, protect_data, camera, None)
+
+        self.device: UnifiCamera = camera
+        self.channel = channel
+        self._secure = secure
         self._disable_stream = disable_stream
-        self._stream_source = None if disable_stream else self._device_data["rtsp"]
         self._last_image = None
+        self._async_set_stream_source()
+        if self._secure:
+            self._attr_unique_id = (
+                f"{self.device.id}_{self.device.mac}_{self.channel.id}"
+            )
+        else:
+            self._attr_unique_id = (
+                f"{self.device.id}_{self.device.mac}_{self.channel.id}_insecure"
+            )
```

Comment:
Do we need both `device.id` and `device.mac`?  mac should be enough?

Reply:
`device.id`  would probably work better hear since the MAC address is unique to the device, which other integrations could use (unless unique IDs for entities are namespace to the integration?). Also, changing it here would make the camera entities inconsistent with every other entity since they are all `id_mac` and we cannot really change those now (that would be a major breaking change).

Reply:
unique ids are name spaced to the integration and platform

https://github.com/home-assistant/core/blob/dev/homeassistant/components/homekit/aidmanager.py#L35

Reply:
We can always migrate them later. 
This one isn't something we need to worry about now.

Reply:
Yeah, I can add the migrate as a follow up PR. Do you have the docs on how to migrate? If I can do that, I can _remove_ the breaking change for the new cameras.

Reply:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/powerwall/__init__.py#L48


and a more complex one


https://github.com/home-assistant/core/blob/dev/homeassistant/components/tplink/migration.py

Reply:
Do you want to do that in this PR?  If so I'll hold off testing since the new entities would be created if I do

Reply:
Nah, I will do it in a follow up PR. This one is already pretty large as is. As of right now only the main camera entities will be broken if you install this version on a HA instance that was running 0.10.

Reply:
The migration _should_ be to test. Either the entities are orphaned or not. So, it should in comparison be a super simple PR to test.

Reply:
Ideally the existing ones get migrated before the new ones are created so users won't loose their custom entity ids and names

Reply:
I was planning on doing the follow up PR before we release the first beta. So only the 3 of us should be affected. Also, merging this PR now should have no effect on the migration. Already have an idea for the logic and it should be the same with or without this PR.